### PR TITLE
Fixing issues after review p12&14

### PR DESCRIPTION
**Pull request by vhumenny**
fixed issues 12 and 14 in the list, Added classes for each command, also added abstract class Command, added class ParsedCommand, moved the logic of executing commands from method parse() to method executeCommand().

### CODE REVIEW

File: src/main/java/commands/CommandAdd.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+@Getter
+public class CommandAdd extends Command {
+
+    private final Pattern regex = (Pattern.compile("^(add) (" +
+            createRegExValues(CartCommandParser.getProducts()) + ") ([0-9]+)"));
+
+    public Boolean matches(String text) {
+        return regex.matcher(text).find();
+    }
```
Comment (horodniuk) [inline src/main/java/commands/CommandAdd.java:18]: Method matches repeated in each command class, can be moove this method in  to class Command

File: src/main/java/commands/CommandAdd.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+import cart.CartCommandParser;
+import lombok.Getter;
+
+import java.util.regex.Pattern;
+
+import static cart.CartCommandParser.createRegExValues;
+
+@Getter
+public class CommandAdd extends Command {
+
```
Comment (horodniuk) [inline src/main/java/commands/CommandAdd.java:12]: Add comment or example for regular expression to make it clear

File: src/main/java/runner/TextModeRunner.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        CartCommandParser cartCommandParser = new CartCommandParser(cart);
+        Optional<ParsedCommand> parsedCommandOptional = cartCommandParser.parse(line);
+        if (parsedCommandOptional.isEmpty())
+            System.out.println("Unknown command, try again, for example \"add beer 5\"");
+        else {
+            Command command = parsedCommandOptional.get().getCommand();
+            List<String> arguments = parsedCommandOptional.get().getArguments();
+            switch (command.getClass().getSimpleName()) {
+                case "CommandDiscount" -> {
+                    cart.applyDiscount(cartCommandParser.parseDiscount(arguments.get(1)), arguments.get(2));
```
Comment (horodniuk) [inline src/main/java/runner/TextModeRunner.java:35]: add comment or example arguments.get(1) and arguments.get(2)

**Review by horodniuk — CHANGES_REQUESTED** [review]

File: src/main/java/commands/CommandAdd.java (hunk 1 above)
Comment (vhumenny) [reply src/main/java/commands/CommandAdd.java:18]: fixed

File: src/main/java/commands/CommandAdd.java (hunk 2 above)
Comment (vhumenny) [reply src/main/java/commands/CommandAdd.java:12]: fixed

File: src/main/java/runner/TextModeRunner.java (hunk 3 above)
Comment (vhumenny) [reply src/main/java/runner/TextModeRunner.java:35]: fixed

**Pull request merged**